### PR TITLE
added styles using tailwindCss, delete not working

### PR DESCRIPTION
**Pull request by ijayhub**
## Description
I've improved the homepage to give it a more polished and professional look. Now, users can easily highlight and keep track of specific collection IDs (lists) that they want to focus on.

## Related Issue
closes #35

## Acceptance Criteria

 Make everything centered
 Remove "Hello from the home (/) page!"
 Change "Name of shopping list:" to "Create a shopping list"
 Add a button to delete the shopping list
 Create a function from the firebase documentation to delete a shopping list/document- not yet
 I want the design to look professional.
 Highlight the shopping list that is selected

## Type of Changes

`new feature` and ` UI enhancement` .

### Before
before- smart shopping list

### After
FireShot Capture 005 - Smart Shopping List - localhost

## Testing Steps / QA Criteria
- Users should be greeted with a visually appealing site upon arrival.
- Creating a list should result in the ability to highlight any list upon clicking it.
- Click on a join button if the collection list already exists.

### CODE REVIEW

**Comment by IamHenryOkeke** [comment]
UI Updates looks great and working fine. Great work

**Review by IamHenryOkeke — APPROVED** [review]
UI Updates looks great and working fine. Great work

**Review by jmahamed — CHANGES_REQUESTED** [review]
I think it looks good, the list can be a bit more centered. I think the background should be just one solid color like #A2A8D3

I think a better title is "Create a shopping list"

button can be "Create new list"

and last button can be "Edit existing list"

File: src/api/firebase.js [hunk 1]
```diff
@@ -224,6 +225,18 @@ export async function deleteItem(listPath, itemId) {
...
+export async function deleteList(listPath, email) {
+	const docRef = doc(db, listPath);
+	console.log(docRef);
+	const userDocRef = doc(db, 'users', email);
+
+	await deleteDoc(docRef);
+
+	await updateDoc(userDocRef, {
+		sharedLists: arrayRemove(docRef),
+	});
```
Comment (redapy) [inline src/api/firebase.js:238]: Nicely done 💥✨! we don't want the user to have a shared list that does not exist 👍

File: src/components/SingleList.jsx [hunk 2]
```diff
@@ -1,13 +1,36 @@
...
-	}
+	};
+
+	// Function to handle deleting a list
+	const handleDelete = async () => {
+		const listPath = '/' + path;
+		await deleteList(listPath, email);
+		localStorage.setItem('tcl-shopping-list-path', '');
+		window.location.reload();
+	};
```
Comment (redapy) [inline src/components/SingleList.jsx:19]: it would be better to comment out this as it is not working

File: src/components/SingleList.jsx [hunk 3]
```diff
@@ -1,13 +1,36 @@
...
-			<button onClick={handleClick}>{name}</button>
-		</li>
+		<div
+			className={`flex items-center justify-between ${selected ? 'selected' : ''}`}
+		>
+			<li className="SingleList">
+				<button onClick={handleClick}>{name}</button>
+			</li>
+			<hr />
+			<MdOutlineDeleteForever className="text-red-700" onClick={handleDelete} />
```
Comment (redapy) [inline src/components/SingleList.jsx]: The delete button should be commented out as well

File: src/views/Home.css [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	background-color: #113767;
+	@apply text-white p-3 rounded-xl shadow-md mb-8;
+}
+:hover .btn {
+	background-color: #38598b;
+}
+.exit-btn {
+	background-color: yellow;
+	color: black;
+	@apply p-3 rounded-xl shadow-md;
```
Comment (redapy) [inline src/views/Home.css]: nice use of the `@apply` directive to make your code more readable  👍

File: src/views/Home.jsx [hunk 5]
```diff
@@ -1,11 +1,12 @@
...
 import './Home.css';
-import { createList } from '../api/firebase';
+import { createList, deleteList } from '../api/firebase';
 import { useState } from 'react';
-import { useNavigate } from 'react-router-dom';
+import { Link, useNavigate } from 'react-router-dom';
 
 export function Home({ data, setListPath, userId, userEmail }) {
 	const [name, setName] = useState('');
+	const [selectedList, setSelectedList] = useState(null); // State to track the selected list
```
Comment (redapy) [inline src/views/Home.jsx]: You do not need this additional state here.  We already have the `listPath` in the `App`, you can pass it as props and use it.

File: src/views/Home.jsx [hunk 6]
```diff
@@ -31,37 +32,53 @@ export function Home({ data, setListPath, userId, userEmail }) {
...
 			<ul>
 				{data?.map((item) => (
 					<SingleList
 						key={item.path}
 						name={item.name}
 						path={item.path}
-						setListPath={setListPath}
+						setListPath={handleListClick}
+						deleteList={deleteList} //  the deleteList function is being passed as a prop
+						selected={selectedList === item.path} // Checks if the list is selected
```
Comment (redapy) [inline src/views/Home.jsx]: as I said above instead of the `selectedList`, use the `lisrPath` from the parent.

**Review by redapy — APPROVED** [review]
Everything is working as expected, great work @ijayhub :rocket: ! I left some non-blocking notes

File: src/api/firebase.js (hunk 1 above)
Comment (ijayhub) [reply src/api/firebase.js:238]: okay changed that

File: src/views/Home.jsx (hunk 5 above)
Comment (ijayhub) [reply src/views/Home.jsx]: okay... thank you

File: src/components/SingleList.jsx (hunk 2 above)
Comment (ijayhub) [reply src/components/SingleList.jsx:19]: it is working now

File: src/components/SingleList.jsx (hunk 3 above)
Comment (ijayhub) [reply src/components/SingleList.jsx]: check again working now

File: src/views/Home.css (hunk 4 above)
Comment (ijayhub) [reply src/views/Home.css]: thank u

File: src/views/Home.jsx (hunk 6 above)
Comment (ijayhub) [reply src/views/Home.jsx]: okay

**Comment by ijayhub** [comment]
> I think it looks good, the list can be a bit more centered. I think the background should be just one solid color like #A2A8D3
> 
> I think a better title is "Create a shopping list"
> 
> button can be "Create new list"
> 
> and last button can be "Edit existing list"

Answer:
 1. It is centers check on small screen
2. I just wanted d homepage visually appealing last demo MJ mention that it was all blue.
3. I updated d U.I
4. For d last button it is just for beauty have changed d text description.

File: src/components/SingleList.jsx (hunk 2 above)
Comment (redapy) [reply src/components/SingleList.jsx:19]: yees, great!!!!!!!!

File: src/views/Home.jsx (hunk 5 above)
Comment (ijayhub) [reply src/views/Home.jsx]: Fixed

File: src/views/Home.jsx (hunk 6 above)
Comment (ijayhub) [reply src/views/Home.jsx]: Fixed

File: src/views/Home.jsx [hunk 7]
```diff
@@ -1,42 +1,31 @@
...
 	};
 
 	const handleChange = (e) => {
 		setName(e.target.value);
 	};
-	// to create a highlighted list
-	const handleListClick = (path) => {
-		setListPath(path);
-		setSelectedList(path); // this update and highlight the list path
-	};
```
Comment (redapy) [inline src/views/Home.jsx]: you should keep the `handleListCLick`, only remove the duplicate `setSelectedList(path)`

File: src/views/Home.jsx [hunk 8]
```diff
@@ -66,9 +55,8 @@ export function Home({ data, setListPath, userId, userEmail }) {
 						name={item.name}
 						email={userEmail}
 						path={item.path}
-						setListPath={handleListClick}
-						selected={selectedList === item.path} // Checks if the list is selected
-						className="SingleList"
+						setListPath={setListPath}
+						listPath={listPath} // Pass listPath to SingleList
 					/>
```
Comment (redapy) [inline src/views/Home.jsx:60]: no need to pass the listPath further, just use it here.
selected={listpath === item.path}

**Review by redapy — CHANGES_REQUESTED** [review]

File: src/views/Home.jsx (hunk 8 above)
Comment (ijayhub) [reply src/views/Home.jsx:60]: I dont really understand...

Comment (redapy) [reply src/views/Home.jsx:60]: before you were using another duplicated state `selectedList` to check if the list is selected like this :
`selected={selectedList === item.path}`. 

Now you will use the `listPath`, that's it:
`selected={listpath === item.path}`

**Review by jmahamed — APPROVED** [review]

**Pull request merged**